howl/data/dataset/labeller.py

```diff
@@ -17,26 +18,119 @@ class PhoneticFrameLabeler(FrameLabeler):
     def __init__(self, pronounce_dict: PronunciationDictionary, phrases: List[PhonePhrase]):
         self.pronounce_dict = pronounce_dict
         self.phrases = phrases
+        punctuation_to_replace = str.maketrans(
+            {"‘": "'", "’": "'", "”": '"', "“": '"', "—": "-", "ä": "a", "ö": "o", "ō": "o", "é": "e", "à": "a"}
+        )
+        punctuation_to_remove = str.maketrans({key: None for key in string.punctuation})
+        # First transformation is None because we want to process the original word first
+        self.punctuation_transforms = [None, punctuation_to_replace, punctuation_to_remove]
+
+    def transform(self, original_word: str) -> PhonePhrase:
+        """Transform the word into list of Phones
+
+        TODO:
+        the function attempts to find list of the longest phone phrase sequences
+        However, this is not ideal for some edge cases.
```

Comment:
Do these types of cases come up often?

Reply:
I am not so sure.
we will need to manually check some words and see how things get parsed.